stacktrace.go

```diff
@@ -113,8 +102,55 @@ type programCounters struct {
 	pcs []uintptr
 }
 
-func newProgramCounters(size int) *programCounters {
-	return &programCounters{make([]uintptr, size)}
+var _stacktracePool = sync.Pool{
+	New: func() interface{} {
+		return &programCounters{make([]uintptr, 64)}
+	},
+}
+
+func newProgramCounters() *programCounters {
+	return _stacktracePool.Get().(*programCounters)
+}
+
+func (pcs *programCounters) Release() {
+	_stacktracePool.Put(pcs)
+}
+
+// Callers captures program counters of callers of the current function.
+//
+// skip specifies the number of functions to skip.
+// limit specifies how deep into the stack we go, zero meaning unlimited.
+func (pcs *programCounters) Callers(skip int, limit int) []uintptr {
+	pc := pcs.pcs
+	if limit > 0 {
+		// The shortened slice will never be placed back into the
+		// pool, but the longer one will.
+		if limit < len(pc) {
+			pc = pc[:limit]
+		} else {
+			pc = make([]uintptr, limit)
+			pcs.pcs = pc
+		}
+	}
+
+	var numFrames int
+	for {
+		// Skip the call to runtime.Callers and programCounter.Callers
+		// so that the program counters start at our caller.
+		numFrames = runtime.Callers(skip+2, pc)
```

Comment:
can do `numFrames := ` since we don't use the value outside of the for loop and always replace it